File: app/src/main/java/com/github/pockethub/android/ui/item/commit/CommitFileHeaderItem.java
package com.github.pockethub.android.ui.item.commit;

import android.content.Context;
import android.content.res.Resources;
import android.support.annotation.NonNull;
import android.support.v4.app.FragmentActivity;
import android.view.View;
import android.widget.TextView;

import com.github.pockethub.android.R;
import com.github.pockethub.android.ui.StyledText;
import com.github.pockethub.android.ui.item.BaseDataItem;
import com.github.pockethub.android.ui.item.BaseViewHolder;
import com.meisolsson.githubsdk.model.GitHubFile;

import java.text.NumberFormat;

import butterknife.BindView;

public class CommitFileHeaderItem
        extends BaseDataItem<GitHubFile, CommitFileHeaderItem.ViewHolder> {

    private final int addTextColor;
    private final int removeTextColor;

    public CommitFileHeaderItem(Context context, GitHubFile file) {
        super(null, file, file.sha().hashCode());

        Resources resources = context.getResources();
        addTextColor = resources.getColor(R.color.diff_add_text);
        removeTextColor = resources.getColor(R.color.diff_remove_text);
    }

    @Override
    public void bind(@NonNull ViewHolder holder, int position) {
        GitHubFile file = getData();
        String path = file.filename();

        int lastSlash = path.lastIndexOf('/');
        if (lastSlash != -1) {
            holder.filename.setText(path.substring(lastSlash + 1));
            holder.folder.setText(path.substring(0, lastSlash + 1));
            holder.folder.setVisibility(View.VISIBLE);
        } else {
            holder.filename.setText(path);
            holder.folder.setVisibility(View.GONE);
        }

        NumberFormat numberFormat = NumberFormat.getIntegerInstance();

        StyledText stats = new StyledText();
        stats.foreground('+', addTextColor);
        stats.foreground(numberFormat.format(file.additions()), addTextColor);
        stats.append(' ').append(' ').append(' ');
        stats.foreground('-', removeTextColor);
        stats.foreground(numberFormat.format(file.deletions()), removeTextColor);
        holder.stats.setText(stats);
    }

    @Override
    public int getLayout() {
        return R.layout.commit_diff_file_header;
    }

    @NonNull
    @Override
    public ViewHolder createViewHolder(@NonNull View itemView) {
        return new ViewHolder(itemView);
    }

    public class ViewHolder extends BaseViewHolder {

        @BindView(R.id.tv_name) TextView filename;
        @BindView(R.id.tv_folder) TextView folder;
        @BindView(R.id.tv_stats) TextView stats;

        public ViewHolder(@NonNull View rootView) {
            super(rootView);
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/commit/CommitFileHeaderItem.kt
package com.github.pockethub.android.ui.item.commit

import android.content.Context
import android.view.View
import androidx.core.text.buildSpannedString
import androidx.core.text.color
import com.github.pockethub.android.R
import com.meisolsson.githubsdk.model.GitHubFile
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.commit_diff_file_header.*
import java.text.NumberFormat

class CommitFileHeaderItem(
        context: Context,
        val file: GitHubFile
) : Item(file.sha()!!.hashCode().toLong()) {

    private val addTextColor = context.resources.getColor(R.color.diff_add_text)
    private val removeTextColor = context.resources.getColor(R.color.diff_remove_text)

    override fun getLayout() = R.layout.commit_diff_file_header

    override fun bind(holder: ViewHolder, position: Int) {
        val path = file.filename()

        val lastSlash = path!!.lastIndexOf('/')
        if (lastSlash != -1) {
            holder.tv_name.text = path.substring(lastSlash + 1)
            holder.tv_folder.text = path.substring(0, lastSlash + 1)
            holder.tv_folder.visibility = View.VISIBLE
        } else {
            holder.tv_name.text = path
            holder.tv_folder.visibility = View.GONE
        }

        val numberFormat = NumberFormat.getIntegerInstance()

        holder.tv_stats.text = buildSpannedString {
            color(addTextColor) {
                append("+${numberFormat.format(file.additions())}")
            }
            append("   ")
            color(removeTextColor) {
                append("-${numberFormat.format(file.deletions())}")
            }
        }
    }
}
